components/external_media_links/external_media_link.lua

```diff
@@ -0,0 +1,271 @@
+local Page = require('Module:Page')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local ExternalMediaLink = {}
+
+local MAXIMUM_VALUES = {
+	subjects = 20,
+	organisations = 5,
+	authors = 5,
+}
+local DEFAULT_LANGUAGE = 'en'
+local NON_BREAKING_SPACE = '&nbsp;'
+
+function ExternalMediaLink.run(args)
+	ExternalMediaLink._fallBackArgs(args)
+
+	if Logic.nilOr(Logic.readBoolOrNil(args.storage), true)
+		and not Logic.readBool(Variables.varDefault('disable_LPDB_storage'))
+		and not Logic.readBool(Variables.varDefault('disable_SMW_storage')) then
+
+		ExternalMediaLink._store(args)
+	end
+
+	mw.ext.TeamLiquidIntegration.add_category('Pages with ExternalMediaLinks')
+	if not Logic.nilOr(Logic.readBoolOrNil(args.display), true) then
+		return
+	end
+
+	return ExternalMediaLink._display(args)
+end
+
+function ExternalMediaLink._fallBackArgs(args)
+	args.by_link1 = args.by_link1 or args.by_link
+	args.by1 = args.by1 or args.by
+	args.subject1 = args.subject1 or args.subject or args.player or args.interviewee
+	args.subject_organization1 = args.subject_organization
+
+	for subjectIndex = 2, MAXIMUM_VALUES.subjects do
+		args['subject' .. subjectIndex] = args['subject' .. subjectIndex] or args['player' .. subjectIndex]
+	end
+end
+
+function ExternalMediaLink._store(args)
+	local lpdbData = {
+		date = args.date,
+		language = args.language or DEFAULT_LANGUAGE,
+		title = args.title,
+		translatedtitle = args.trans_title,
+		link = args.link,
+		publisher = args.of,
+		type = args.type,
+	}
+
+	local authors = {}
+	for authorIndex, author in Table.iter.pairsByPrefix(args, 'by') do
+		authors['author' .. authorIndex] =
+			mw.ext.TeamLiquidIntegration.resolve_redirect(args['by_link' .. authorIndex] or author)
+		authors['author' .. authorIndex .. 'dn'] = author
+	end
+	-- set a maximum for authors due to the same being used in queries
+	assert(Table.size(authors) <= MAXIMUM_VALUES.authors,
+		'Maximum Value of authors (' .. MAXIMUM_VALUES.authors .. ') exceeded')
+	lpdbData.authors = mw.ext.LiquipediaDB.lpdb_create_json(authors)
+
+	local extradata = {
+		translation = args.translation,
+		translator = args.translator,
+		event = args.event,
+		event_link = args['event-link'],
+		subject_organization = args.subject_organization1, --legacy
+	}
+
+	local orgs = {}
+	for orgIndex, org in Table.iter.pairsByPrefix(args, 'subject_organization') do
+		orgs['subject_organization' .. orgIndex] = mw.ext.TeamLiquidIntegration.resolve_redirect(org)
+	end
+	-- set a maximum for orgs due to the same being used in queries
+	assert(Table.size(orgs) <= MAXIMUM_VALUES.organisations,
+		'Maximum Value of organisations subjects (' .. MAXIMUM_VALUES.organisations .. ') exceeded')
+
+	local subjects = {}
+	for subjectIndex, subject in Table.iter.pairsByPrefix(args, 'subject') do
+		subjects['subject' .. subjectIndex] = mw.ext.TeamLiquidIntegration.resolve_redirect(subject)
```

Comment:
Same as with orgs, could/should store dn

Reply:
there is no display for them they only get stored (and conditioned on)